docs/remote/containers.md

```diff
@@ -231,6 +235,36 @@ Selecting the **Remote-Containers: Add Development Container Configuration Files
 
 To learn more about creating `devcontainer.json` files, see [Create a Development Container](/docs/remote/create-dev-container.md).
 
+## Dev container features
+
+Dev container features provide a smooth path for customizing your container definitions.
+
+When you use **Remote-Containers: Add Development Container Configuration Files**, you're presented a list of scripts to customize the existing dev container configurations, such as installing Git or the Azure CLI:
+
+![Dev container features in Command Palette](images/containers/container-features.png)
+
+When you rebuild and reopen in your container, the features you selected will be available in your `devcontainer.json`:
+
+```json
+"features": {
+		"github-cli": "latest"
+	}
+```
+
+The features are sourced from the [script library](https://github.com/microsoft/vscode-dev-containers/tree/main/script-library) in the vscode-dev-containers repo.
+
+## Pre-building
+
+You can pre-build images with the tools you need rather than newly creating and building images for each project. Using pre-built images can be simpler and allows you to pin to a specific version of tools, avoiding potential breaks. Pre-building is especially valuable in CI processes.
+
+You can use the [devcontainer CLI](/remote/devcontainer-cli.md) to facilitate pre-builds.
+
+The process could be as follows:
+* Create a source code repository
+* Create a dev container configuration, customizing as you wish (including [features](#Dev-container-features))
+* Use the `devcontainer CLI` to build your image (the `devcontainer CLI` supports building images with features)
+* Push your image: `docker push <your_image_name>`
+
```

Comment:
I'd suggest a code snippet illustrating the commands and mentioning that you can push your image to a container registry like the Azure Continer Registry or GitHub Container Registry (with links to docs for each).

```
$ devcontainer build --image-name your-registry.azurecr.io/your-image-name
$ docker push your-registry.azurecr.io/your-image-name
```

We also need to tell people how to install the dev dcontainer CLI w/o VS Code now that this is possible (though that could happen over on the CLI page.)

Reply:
I have added basic installation instructions in the CLI's README.md. These will show up on the package's page next time we publish the package. https://github.com/microsoft/vscode-remote-containers/blob/main/dev-containers-user-cli/README.md

Reply:
Updated this section with example commands like Chuck lists, more links to docs, and info from the CLI's readme like Christof mentioned. Let me know what you think!

Reply:
@bamurtaugh So based on today's discussion, we'll need to update the instructions for installing the CLI, but generally something along these lines makes sense.

I'd suggest we move the general install instructions for the CLI into the existing dev container CLI document we have that describes the command you can use.  We could have "install using VS Code" and "Install from the command line" as two install options